test/integration/EnrolmentStoreProxyConnectorSpec.scala

```diff
@@ -162,5 +165,159 @@ class EnrolmentStoreProxyConnectorSpec extends IntegrationTestsSpec with ScalaFu
+
+        val enrolmentStoreProxyResponse = Json.parse("""{
+            |    "principalGroupIds": [
+            |       "groupId1",
+            |       "groupId2"
+            |    ]
+            |}""".stripMargin)
+
+        EnrolmentStoreProxyService.stubTheEnrolmentStoreProxyResponse(
+          s"/enrolment-store-proxy/enrolment-store/enrolments/${es1Request.enrolment}/groups?type=${es1Request.queryType.value}",
+          enrolmentStoreProxyResponse.toString(),
+          OK
+        )
+
+        val result = enrolmentStoreProxyConnector.queryGroupsWithAllocatedEnrolment(es1Request)
+
+        val expectedResponse =
+          ES1Response(Some(Seq("groupId1", "groupId2")), None)
+
+        result.futureValue mustBe expectedResponse
+      }
+
+      "enrolment store proxy respond with status OK and delegated groupIds" in {
+        val es1Request = ES1Request(Service.withName("atar").get, "GB123456789123")
+
+        val enrolmentStoreProxyResponse = Json.parse("""{
+            |    "delegatedGroupIds": [
+            |       "groupId3",
+            |       "groupId4"
+            |    ]
+            |}""".stripMargin)
+
+        EnrolmentStoreProxyService.stubTheEnrolmentStoreProxyResponse(
+          s"/enrolment-store-proxy/enrolment-store/enrolments/${es1Request.enrolment}/groups?type=${es1Request.queryType.value}",
+          enrolmentStoreProxyResponse.toString(),
+          OK
+        )
+
+        val result = enrolmentStoreProxyConnector.queryGroupsWithAllocatedEnrolment(es1Request)
+
+        val expectedResponse =
+          ES1Response(None, Some(Seq("groupId3", "groupId4")))
+
+        result.futureValue mustBe expectedResponse
+      }
+
+      "enrolment store proxy respond with status NO_CONTENT" in {
+        val es1Request = ES1Request(Service.withName("atar").get, "GB123456789123")
+
+        EnrolmentStoreProxyService.stubTheEnrolmentStoreProxyResponse(
+          s"/enrolment-store-proxy/enrolment-store/enrolments/${es1Request.enrolment}/groups?type=${es1Request.queryType.value}",
+          "",
+          NO_CONTENT
+        )
+
+        val result = enrolmentStoreProxyConnector.queryGroupsWithAllocatedEnrolment(es1Request)
+
+        val expectedResponse = ES1Response(None, None)
+
+        result.futureValue mustBe expectedResponse
+      }
+    }
+
+    "throw an exception" when {
+
+      "status is 400 (BAD_REQUEST) with message INVALID_ENROLMENT_KEY" in {
+
+        val es1Request = ES1Request(Service.withName("atar").get, "GB123456789123")
+
+        val enrolmentStoreProxyResponse =
+          Json.parse("""{
+            |    "code":"INVALID_ENROLMENT_KEY",
+            |    "message":"The enrolment key provided did not match the specification"
+            |}""".stripMargin)
+
+        EnrolmentStoreProxyService.stubTheEnrolmentStoreProxyResponse(
+          s"/enrolment-store-proxy/enrolment-store/enrolments/${es1Request.enrolment}/groups?type=${es1Request.queryType.value}",
```

Comment:
@PatrykRudnicki : can we send an incorrect service name to make it more obvious why we are returning 400 ?

Reply:
Yes, good point.

Will do the same for incorrect type

Reply:
Changed to incorrect service and used incorrect type for the second example.

I needed to mock it, because ES1QueryType is a sealed abstract class, so there is no other way to create incorrect type.